src/helium_follower_service.erl

```diff
@@ -56,9 +60,80 @@ subnetwork_last_reward_height(Ctx, Req) ->
     subnetwork_last_reward_height(Chain, Ctx, Req).
 
 -spec active_gateways(follower_pb:follower_gateway_stream_req_v1_pb(), grpcbox_stream:t()) ->
-    {ok, grpcbox_stream:t()} | grpcbox_stream:grpc_error_response().
+        {ok, grpcbox_stream:t()} | grpcbox_stream:grpc_error_response().
+active_gateways(#follower_gateway_stream_req_v1_pb{batch_size = BatchSize} = _Msg, StreamState) when BatchSize =< ?GW_STREAM_BATCH_SIZE ->
+    #{chain := Chain} =  grpcbox_stream:stream_handler_state(StreamState),
+    case Chain of
+        undefined ->
+            lager:debug("chain not ready, returning error response for msg ~p", [_Msg]),
+            {grpc_error, {grpcbox_stream:code_to_status(14), <<"temporarily unavailable">>}};
+        _ ->
+            Ledger = blockchain:ledger(Chain),
+            {ok, Height} = blockchain_ledger_v1:current_height(Ledger),
+
+            InteractiveBlock = case ?get_var(?hip17_interactivity_blocks, Ledger) of
+                                   {ok, V} -> V;
+                                   {error, not_found} -> 0
+                               end,
+            {NumGateways, FinalGws, StreamState1} =
+                blockchain_ledger_v1:cf_fold(
+                    active_gateways,
+                    fun({Addr, BinGw}, {CountAcc, GwAcc, StreamAcc}) ->
+                        Gw = blockchain_ledger_gateway_v2:deserialize(BinGw),
+                        Loc = blockchain_ledger_gateway_v2:location(Gw),
+                        LastBeacon = case blockchain_ledger_v1:find_gateway_last_beacon(Addr, Ledger) of
+                                         {ok, LB} -> LB;
+                                         {error, _} -> undefined
+                                     end,
+                        case Loc /= undefined andalso (LastBeacon /= undefined andalso (Height - LastBeacon) =< InteractiveBlock) of
+                            true ->
+                                Region = case blockchain_region_v1:h3_to_region(Loc, Ledger) of
+                                             {ok, R} -> normalize_region(R);
+                                             _ -> undefined
+                                         end,
+                                GwResp = #follower_gateway_resp_v1_pb{
+                                          height = Height,
+                                          result = {info, #gateway_info_pb{
+                                              address = Addr,
+                                              location = h3:to_string(Loc),
```

Comment:
why does the follower convert location to a h3 string. Was there a hard reason for doing that ?

Reply:
I have no idea why we went with a string in the proto; it’s seems like we’re always converting to a u64

Reply:
This was historical because the h3 locations were used/returned by javascript which had issues with all of the u64 bits being used.. The convention stuck